interbtc-spec/docs/source/spec/vault-registry.rst

```diff
@@ -823,10 +837,29 @@ Specification
 
 *Postconditions*
 
-* Function :ref:`reward_withdrawStake` MUST complete successfully - parameterized by ``vault`` and ``vault.issuedTokens``.
-* The liquidation vault's ``issuedTokens``, ``toBeIssuedTokens`` and ``toBeRedeemedTokens`` MUST be increased by the respective amounts in the vault.
-* The vault's ``issuedTokens`` and ``toBeIssuedTokens`` MUST be set to 0.
-* Collateral MUST be moved from the vault to the liquidation vault: an amount of ``confiscatedCollateral - confiscatedCollateral * (toBeRedeemedTokens / (toBeIssuedTokens + issuedTokens))`` is moved, where ``confiscatedCollateral`` is the minimum of the vault's ``backingCollateral`` (as calculated via :ref:`computeStakeAtIndex`) and ``SecureCollateralThreshold`` times the equivalent worth of the amount of tokens it is backing.
+* ``usedCollateral`` MUST be calculated as ``exchangeRate * (issuedTokens + toBeIssuedTokens)) * secureCollateralThreshold``.
+* ``usedCollateral`` MUST be set to ``backingCollateral`` if ``backingCollateral < usedCollateral``.
+* ``usedTokens`` MUST be calculated as ``issuedTokens - toBeIssuedTokens``.
+* ``toBeLiquidated`` MUST be calculated as ``(usedCollateral * (usedTokens - toBeRedeemedTokens)) / usedTokens``.
+* ``remainingCollateral`` MUST be calculated as ``usedCollateral - toBeLiquidated``.
+* Function :ref:`reward_withdrawStake` MUST complete successfully - parameterized by ``vault`` and ``issuedTokens``.
+* Function :ref:`staking_withdrawStake` MUST complete successfully - parameterized by ``vault`` and ``remainingCollateral``.
```

Comment:
I think this is wrong. The call to `slash_to_liquidation_vault` is with `to_be_liquidated` as amount, which then does some calculation on.

As a side note, having the naming mismatch makes it a lot harder to check correctness..

Reply:
I'm not sure exactly what you think is wrong, could you clarify?

Reply:
The argument to `staking_withdrawStake`. I don't think it's called with `remainingCollateral`, but due to the confusing naming I am not 100% sure.

Reply:
I'll merge and resolve after the audit fixes - will harmonize the naming to make this easier to check.